spec/spec_helper.rb

```diff
@@ -3,27 +3,26 @@
   SimpleCov.start
 end
 
-require 'vcr'
-require 'cgi'
+Dir[Rails.root.join("spec/shared/**/*.rb")].each { |f| require f }
+Dir[File.join(__dir__, "support/**/*.rb")].each { |f| require f }
+
+require "manageiq-providers-google"
 
 VCR.configure do |config|
   config.ignore_hosts 'codeclimate.com' if ENV['CI']
   config.cassette_library_dir = File.join(ManageIQ::Providers::Google::Engine.root, 'spec/vcr_cassettes')
 
-  # Set your config/secrets.yml file
-  secrets = Rails.application.secrets
-
   # Looks for provider subkeys you set in secrets.yml. Replace the values of
   # those keys (both escaped or unescaped) with some placeholder text.
+  require "cgi"
+  secrets = Rails.application.secrets
   secrets.each_key do |provider|
     next if %i(secret_key_base secret_token).include?(provider) # Defaults
+
     cred_hash = secrets.public_send(provider)
     cred_hash.each do |key, value|
       config.filter_sensitive_data("#{provider.upcase}_#{key.upcase}") { CGI.escape(value) }
       config.filter_sensitive_data("#{provider.upcase}_#{key.upcase}") { value }
     end
   end
```

Comment:
I feel like this secrets thing should be in the code provider generator.  This way, every plugin can follow the same pattern if this is the one we like.  Also, I'm not sure why config/secrets.yml is not in the .gitignore like I see in the Azure plugin (and I'm not sure why the Azure plugin doesn't have this code).  @agrare What do you think?

Reply:
If not in the generator, then maybe a spec/support/shared file?

Reply:
I don't think `config/secrets.yml` is pluggable so I don't know why azure has `config/secrets.yml` in its `.gitignore`